The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10002785352/job/27648655348) [(plain)](https://github.com/rust-lang/rust/commit/c72df9ddf946e9e21226c03d5e2b373dbe862fc4/checks/27648655348/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:692973e3d937129bcbf40652eb9f2f61becf3332)
Download action repository 'actions/upload-artifact@v4' (SHA:0b2256b8c012f0828dc542b3febcab082c67f72b)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.460   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.481 Collecting boolean-py==4.0
#13 0.490   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.507 Collecting chardet==5.1.0
---
#13 4.315 Building wheels for collected packages: reuse
#13 4.316   Building wheel for reuse (pyproject.toml): started
#13 4.656   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 4.657   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 4.657   Stored in directory: /tmp/pip-ephem-wheel-cache-kakwrttm/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 4.660 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 4.684   Attempting uninstall: setuptools
#13 4.684     Found existing installation: setuptools 59.6.0
#13 4.686     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 4.686     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 4.686     Can't uninstall 'setuptools'. No files were found to uninstall.
#13 5.386 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#13 5.386 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 5.947 Collecting virtualenv
#13 6.002   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 6.227      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 25.5 MB/s eta 0:00:00
#13 6.270 Collecting distlib<1,>=0.3.7
#13 6.279   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 6.327 Collecting filelock<4,>=3.12.2
#13 6.335   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 6.335   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 6.367 Collecting platformdirs<5,>=3.9.1
#13 6.374   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 6.464 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 6.649 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 6.7s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      200640 kB
DirectMap2M:     8187904 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/3d68afc9e821b00d59058abc9bda670b07639955/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-3d68afc9e821b00d59058abc9bda670b07639955-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 28.83s
##[endgroup]
fmt check
##[error]Diff in /checkout/src/tools/compiletest/src/runtest.rs at line 280:
             Ui | MirOpt => false,
             mode => panic!("unimplemented for mode {:?}", mode),
-        if test_should_run {
-            self.run_if_enabled()
-        } else {
-            WillExecute::No
-            WillExecute::No
-        }
+        if test_should_run { self.run_if_enabled() } else { WillExecute::No }
 
 
     fn run_if_enabled(&self) -> WillExecute {
##[error]Diff in /checkout/src/tools/compiletest/src/runtest.rs at line 291:
-        if self.config.run_enabled() {
-            WillExecute::Yes
-            WillExecute::Disabled
-        }
-        }
+        if self.config.run_enabled() { WillExecute::Yes } else { WillExecute::Disabled }
 
 
     fn should_run_successfully(&self, pm: Option<PassMode>) -> bool {
##[error]Diff in /checkout/src/tools/compiletest/src/runtest.rs at line 2451:
     /// The revision, ignored for incremental compilation since it wants all revisions in
     fn safe_revision(&self) -> Option<&str> {
-        if self.config.mode == Incremental {
-            None
-        } else {
-        } else {
-            self.revision
-        }
+        if self.config.mode == Incremental { None } else { self.revision }
 
     /// Gets the absolute path to the directory where all output for the given
     /// Gets the absolute path to the directory where all output for the given
##[error]Diff in /checkout/src/tools/compiletest/src/runtest.rs at line 2679:
     fn charset() -> &'static str {
     fn charset() -> &'static str {
         // FreeBSD 10.1 defaults to GDB 6.1.1 which doesn't support "auto" charset
-        if cfg!(target_os = "freebsd") {
-            "ISO-8859-1"
-            "UTF-8"
-        }
-        }
+        if cfg!(target_os = "freebsd") { "ISO-8859-1" } else { "UTF-8" }
 
     fn run_rustdoc_test(&self) {
     fn run_rustdoc_test(&self) {
##[error]Diff in /checkout/src/tools/compiletest/src/runtest.rs at line 4506:
         for output_file in files {
             println!("Actual {} saved to {}", kind, output_file.display());
-        if self.config.bless {
-            0
-        } else {
-            1
-            1
-        }
+        if self.config.bless { 0 } else { 1 }
 
     fn check_and_prune_duplicate_outputs(
     fn check_and_prune_duplicate_outputs(
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/src/tools/compiletest/src/header/test-auxillary/known_directive.rs" "/checkout/src/tools/compiletest/src/header/test-auxillary/unknown_directive.rs" "/checkout/src/tools/compiletest/src/header/test-auxillary/known_legacy_directive.rs" "/checkout/src/tools/compiletest/src/header/cfg.rs" "/checkout/src/tools/compiletest/src/header/needs.rs" "/checkout/src/tools/compiletest/src/header/tests.rs" "/checkout/src/tools/compiletest/src/runtest.rs" "/checkout/compiler/rustc_target/src/spec/targets/wasm64_unknown_unknown.rs" "/checkout/compiler/rustc_target/src/spec/targets/sparc_unknown_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/sparcv9_sun_solaris.rs" "/checkout/compiler/rustc_target/src/spec/targets/i386_apple_ios.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_pc_windows_gnullvm.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/loongarch64_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/csky_unknown_linux_gnuabiv2hf.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_none_softfloat.rs" "/checkout/compiler/rustc_target/src/spec/targets/wasm32_unknown_emscripten.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_unknown_openbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_netbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv7neon_unknown_linux_musleabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv4t_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv64gc_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_apple_darwin.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32gc_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/xtensa_esp32_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_illumos.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_uwp_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_apple_ios_macabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_l4re_uclibc.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_wrs_vxworks_spe.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_linux_android.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32imafc_esp_espidf.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv5te_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/arm_unknown_linux_musleabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/arm64e_apple_darwin.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_redox.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_apple_watchos_sim.rs" "/checkout/compiler/rustc_target/src/spec/targets/mips_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32imc_esp_espidf.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32gc_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_pc_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_pc_windows_gnullvm.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv4t_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/msp430_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32i_unknown_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_apple_darwin.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv7m_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/mipsel_sony_psp.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32ima_unknown_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_linux_android.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_be_unknown_netbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32im_unknown_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_fuchsia.rs" "/checkout/src/tools/build-manifest/src/manifest.rs" "/checkout/src/tools/build-manifest/src/checksum.rs" "/checkout/src/tools/build-manifest/src/versions.rs" "/checkout/src/tools/build-manifest/src/main.rs" "/checkout/src/tools/llvm-bitcode-linker/src/bin/llvm-bitcode-linker.rs" "/checkout/src/tools/llvm-bitcode-linker/src/lib.rs" "/checkout/src/tools/llvm-bitcode-linker/src/linker.rs" "/checkout/src/tools/llvm-bitcode-linker/src/opt.rs" "/checkout/src/tools/llvm-bitcode-linker/src/target.rs" "/checkout/src/tools/compiletest/src/header/test-auxillary/error_annotation.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Fri Jul 19 05:00:12 UTC 2024
  network time: Fri, 19 Jul 2024 05:00:12 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/tools/compiletest/src/runtest.rs
+++ b/src/tools/compiletest/src/runtest.rs
@@ -280,19 +280,11 @@ impl<'test> TestCx<'test> {
             Ui | MirOpt => false,
             mode => panic!("unimplemented for mode {:?}", mode),
         };
-        if test_should_run {
-            self.run_if_enabled()
-        } else {
-            WillExecute::No
-        }
+        if test_should_run { self.run_if_enabled() } else { WillExecute::No }
     }
 
     fn run_if_enabled(&self) -> WillExecute {
-        if self.config.run_enabled() {
-            WillExecute::Yes
-        } else {
-            WillExecute::Disabled
-        }
+        if self.config.run_enabled() { WillExecute::Yes } else { WillExecute::Disabled }
     }
 
     fn should_run_successfully(&self, pm: Option<PassMode>) -> bool {
@@ -2451,11 +2443,7 @@ impl<'test> TestCx<'test> {
     /// The revision, ignored for incremental compilation since it wants all revisions in
     /// the same directory.
     fn safe_revision(&self) -> Option<&str> {
-        if self.config.mode == Incremental {
-            None
-        } else {
-            self.revision
-        }
+        if self.config.mode == Incremental { None } else { self.revision }
     }
 
     /// Gets the absolute path to the directory where all output for the given
@@ -2679,11 +2667,7 @@ impl<'test> TestCx<'test> {
 
     fn charset() -> &'static str {
         // FreeBSD 10.1 defaults to GDB 6.1.1 which doesn't support "auto" charset
-        if cfg!(target_os = "freebsd") {
-            "ISO-8859-1"
-        } else {
-            "UTF-8"
-        }
+        if cfg!(target_os = "freebsd") { "ISO-8859-1" } else { "UTF-8" }
     }
 
     fn run_rustdoc_test(&self) {
@@ -4506,11 +4490,7 @@ impl<'test> TestCx<'test> {
         for output_file in files {
             println!("Actual {} saved to {}", kind, output_file.display());
         }
-        if self.config.bless {
-            0
-        } else {
-            1
-        }
+        if self.config.bless { 0 } else { 1 }
     }
 
     fn check_and_prune_duplicate_outputs(

